Updating retriever-examples documentation to run validation tests on the provided snippets (#116643)
elastic/elasticsearch: docs/reference/search/rrf.asciidoc

* `from=0, size=2` would return [`1`, `5`] with ranks `[1, 2]`
* `from=2, size=2` would return an empty result set as it would fall outside the available `rank_window_size` results.

==== Aggregations in RRF

> Using the `rrf` retriever, we can also gather aggregations from all its specified sub-retrievers. The aggregations gathered
> are irrespective of the specified `rank_window_size` but instead we refer to the union of the result sets from all sub-retrievers,
> i.e. we collect all matching documents and not just the top `rank_window_size`.

Reviewer: ```suggestion
The `rrf` retriever supports aggregations from all specified sub-retrievers. Important notes about aggregations:

* They operate on the complete result set from all sub-retrievers
* They are not limited by the `rank_window_size` parameter
* They process the union of all matching documents
```

> So for example, assuming that we have the following documents:

Reviewer: ```suggestion
For example, consider the following document set:
```

> [source,python]
> ----
> # doc     |   termA      |    termB    |
> _id: 1 =       foo
> _id: 2 =       foo             bar
> _id: 3 =     aardvark          bar
> _id: 4 =       foo             bar
> ----

Reviewer: ```suggestion
[source,js]
----
{
    "_id": 1, "termA": "foo",
    "_id": 2, "termA": "foo", "termB": "bar",
    "_id": 3, "termA": "aardvark", "termB": "bar",
    "_id": 4, "termA": "foo", "termB": "bar"
}
----
```

Reviewer: maybe json makes more sense here?

// NOTCONSOLE

> And the following `rrf` query with a term aggregation specified on field `termA`:

Reviewer: ```suggestion
Perform a term aggregation on the `termA` field using an `rrf` retriever:
```

[source,js]
----
{
    "retriever": {
        "rrf": {
            "retrievers": [
                {
                    "standard": {
                        "query": {
                            "term": {
                                "termB": "bar"
                            }
                        }
                    }
                },
                {
                    "standard": {
                        "query": {
                            "match_all": { }
                        }
                    }
                }
            ],
            "rank_window_size": 1
        }
    },
    "size": 1,
    "aggs": {
        "termA_agg": {
            "terms": {
                "field": "termA"
            }
        }
    }
}
----
// NOTCONSOLE

> Even though we have a `rank_window_size: 1`, the aggregations will be computed against **all** matching documents from the nested sub-retrievers.

Reviewer: ```suggestion
The aggregation results will include *all* matching documents, regardless of `rank_window_size`.
```

> So, the expected aggs would be:

Reviewer: ```suggestion

```

[source, js]
----
{
    "foo": 3,
    "aardvark": 1
}

----
// NOTCONSOLE

==== Highlighting in RRF

> Similarly to above, we can also add <<highlighting, highlight snippets>> to the `rrf` retriever's results. Highlighted snippets are computed based

Reviewer: ```suggestion
Using the `rrf` retriever, you can add <<highlighting, highlight snippets>> to show relevant text snippets in your search results. Highlighted snippets are computed based
```

on the matching text queries defined on the sub-retrievers.

> NOTE:: Highlighting on vector fields, using either the `knn` retriever or a `knn` query, is not supported at the moment.

Reviewer: ```suggestion
IMPORTANT: Highlighting on vector fields, using either the `knn` retriever or a `knn` query, is not supported.
```

A more specific example of highlighting in RRF can also be found in the <<retrievers-examples-highlighting-retriever-results, retrievers examples>> page.

==== Inner hits in RRF

> Computing <<inner-hits, inner hits>> is now also an option for RRF. We can specify inner hits as part of a nested sub-retriever, which
> will be propagated to the top level parent retriever. Actual inner hit computation will take place only at end of `rrf` retriever's evaluation on the top matching documents,
> and not as part of the query execution of the nested sub-retrievers.

Reviewer: ```suggestion
The `rrf` retriever supports <<inner-hits,inner hits>> functionality, allowing you to retrieve 
related nested or parent/child documents alongside your main search results. Inner hits can be 
specified as part of any nested sub-retriever and will be propagated to the top-level parent 
retriever. Note that the inner hit computation will take place only at end of `rrf` retriever's 
evaluation on the top matching documents, and not as part of the query execution of the nested 
sub-retrievers.
```
